app/api/inventory/route.ts

```diff
@@ -16,33 +16,125 @@ async function createInventoryItem(data : {
 }
 
 
-async function getInventoryItems() {
+export async function getInventoryItems() {
   return await prisma.inventory.findMany()
 }
 
 
-async function updateInventoryItem(data : {
+export async function updateInventoryItem(data : {
   itemName     : string,
   categoryName : string,
   quantity     : number, 
   units        : string, 
   lastUpdated  : Date
 }) {
-  const { itemName, ...newData } = data;
+  const { itemName, units, ...newData } = data;
   return await prisma.inventory.update({
     where : {
-      itemName : itemName
+      itemName_units : {
+        itemName: itemName,
+        units: units,
+      }
     }, data: { ...newData }
   })
 }
 
 
-async function deleteInventoryItem(deleteItem : string) {
+export async function deleteInventoryItem(data: {
+  itemName : string,
+  units    : string
+}) {
+  const { itemName, units } = data;
+
   return await prisma.inventory.delete({
-    where : {
-      itemName : deleteItem
-    },
-  })
+    where: {
+      itemName_units: {
+        itemName: itemName,
+        units: units,
+      }
+    }
+  });
+}
+
+
+export async function POST(req: NextRequest) {
+    try {
+      const body = await req.json();
+      if (body.lastUpdated && typeof body.lastUpdated === 'string') {
+        body.lastUpdated = new Date(body.lastUpdated);
+      }
+      const result = await createInventoryItem({
+        ...body,
+      });
+      console.log(result);
+      return NextResponse.json({ message: 'Successfully Created', status: 201, data: result })
+    } catch(error) {
+      return NextResponse.json({ message: 'Unexpected Error', status: 500 })
+    }
+}
+
+export async function GET() {
+  try {
+    const result = await getInventoryItems()
+    return NextResponse.json({ message: 'OK', status: 200, data: result })
+  } catch(error) {
+    return NextResponse.json({ message: 'Unexpected Error', status: 500 })
+  }
+}
+
+export async function PUT(req: NextRequest) {
+  try {
+    const body = await req.json();
+
+    // Ensure lastUpdated is a valid Date
+    if (body.lastUpdated && typeof body.lastUpdated === 'string') {
+      body.lastUpdated = new Date(body.lastUpdated);
+    }
+
+    // Ensure the body contains both itemName and units (they are primary keys)
+    const { itemName, units } = body;
+
+    if (!itemName || !units) {
+      console.log(itemName);
+      console.log(units);
+      return NextResponse.json({ message: 'Missing itemName or units', status: 400 });
+    }
```

Comment:
This is some new logic I introduced after Idil & I realized that we needed to remigrate our inventory prisma schema because the primary key needed to include units.

Essentially, I'm extracting the itemName and units from the request body and ensuring that they exist before calling the utility function for updateInventoryItem. This means that every time someone wants to update an inventory item, the backend finds the item based on the `itemName` AND the `units` because inventory items can have different units and we are representing them as separate rows.